packages/atlas/src/providers/transactions/transactions.hooks.ts

```diff
@@ -247,7 +249,7 @@ export const useTransaction = (): HandleTransactionFn => {
         })
 
         /* === transaction was successful, do necessary cleanup === */
-        updateStatus(ExtrinsicStatus.Completed)
+        !onTxSuccess && updateStatus(ExtrinsicStatus.Completed)
```

Comment:
Why do we want to switch the state only when the transaction has NOT succeeded?

Reply:
@attemka maybe I missed the naming here but heres how it'll work: onTxSuccess should run the chained transaction, so I've hidden the success state of the current one, as it could be confusing for the user. 2nd transaction (or the last one, if many) won't have onTxSuccess, so it'll display the final success state. Maybe I should change the propertyName, but I've ran out of ideas tbh.

Reply:
I think at this point we should just introduce an API that will handle performing batch transactions

Reply:
Idk, this feels pretty weird for me and unintuitive

Reply:
Agreed to implement custom modal approach in future iterations.